core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterBuilderService.java

```diff
@@ -191,13 +266,47 @@ public void buildCluster(Long stackId) throws CloudbreakException, ClusterClient
         }
     }
 
-    private void setupProxy(ClusterApi connector, ProxyConfig proxyConfig) {
-        if (proxyConfig != null) {
-            LOGGER.info("proxyConfig is not null, setup proxy for cluster: {}", proxyConfig);
-            connector.clusterSetupService().setupProxy(proxyConfig);
-        } else {
-            LOGGER.info("proxyConfig was not found by proxyConfigCrn");
-        }
+    private String sdxStackCrn(Stack stack) {
```

Comment:
I would rename this and all the other **new** "getter" functions below this to a more descriptive one.
Let's say this from sdxStackCrn -> getSdxStackCrnByStack
Or for example getClusterSetupService -> getClusterSetupServiceByStackId
Or datalakeResources -> getDatalakeResourcesByStack

And so on. 

These are private function be the have to have as descriptive names as possible to avoid unnecessary jumping inside to code (to figure out what am I actually calling)

Reply:
how about prepareSdxStackCrn?